src/neurostatslib/utils.py

```diff
@@ -4,44 +4,263 @@
 from __future__ import annotations
 
 from functools import partial
-from typing import Optional, Union
+from typing import Iterable, List, Optional, Union
 
 import jax
 import jax.numpy as jnp
 import matplotlib.pyplot as plt
 import numpy as np
-from numpy.typing import ArrayLike
+from numpy.typing import ArrayLike, NDArray
 
-# Broadcasted 1d convolution operations.
-# [[n x t],[w]] -> [n x (t - w + 1)]
+# Same trial duration
+# [[r x t x n], [w]] -> [r x (t - w + 1) x n]
```

Comment:
Nitpick: I find `x` confusing when writing the dimensionality like this. It's fine if it's clear to you, but I tend to go with `,` to match numpy convention, so `[[r, t, n], [w]]`